lib/Service/ImageService.php

```diff
@@ -0,0 +1,596 @@
+			$file = $userFolder->get($filePath);
+			if ($file instanceof File) {
+				return $file;
+			}
+		}
+		return null;
+	}
+
+	/**
+	 * Get a user file from file ID
+	 *
+	 * @param int $textFileId
+	 * @param string $userId
+	 * @return Folder|null
+	 * @throws \OCP\Files\InvalidPathException
+	 * @throws \OCP\Files\NotFoundException
+	 * @throws \OCP\Files\NotPermittedException
+	 * @throws \OC\User\NoUserException
+	 */
+	private function getTextFile(int $textFileId, string $userId): ?File {
+		$userFolder = $this->rootFolder->getUserFolder($userId);
+		$textFile = $userFolder->getById($textFileId);
+		if (count($textFile) > 0 && $textFile[0] instanceof File) {
+			return $textFile[0];
+		}
+		return null;
+	}
+
+	/**
+	 * Get file from share token
+	 *
+	 * @param int|null $textFileId
+	 * @param string $shareToken
+	 * @return File|null
+	 * @throws NotFoundException
+	 */
+	private function getTextFilePublic(?int $textFileId, string $shareToken): ?File {
+		// is the file shared with this token?
+		try {
+			$share = $this->shareManager->getShareByToken($shareToken);
+			if ($share->getShareType() === IShare::TYPE_LINK) {
+				// shared file or folder?
+				if ($share->getNodeType() === 'file') {
+					$textFile = $share->getNode();
+					if ($textFile instanceof File) {
+						return $textFile;
+					}
+				} elseif ($share->getNodeType() === 'folder' && $textFileId !== null) {
+					$folder = $share->getNode();
+					if ($folder instanceof Folder) {
+						$textFile = $folder->getById($textFileId);
+						if (count($textFile) > 0 && $textFile[0] instanceof File) {
+							return $textFile[0];
+						}
+					}
+				}
+			}
+		} catch (ShareNotFound $e) {
+			return null;
+		}
+		return null;
+	}
+
+	/**
+	 * Download a file and write it to a resource
+	 *
+	 * @param string $url
+	 * @param $resource
+	 * @param array $params
+	 * @param string $method
+	 * @return array|string[]
+	 */
+	private function simpleDownload(string $url, $resource, array $params = [], string $method = 'GET'): array {
+		$client = $this->clientService->newClient();
+		try {
+			$options = [
+				// does not work with sink if SSE is enabled
+				// 'sink' => $resource,
+				// rather use stream and write to the file ourselves
+				'stream' => true,
+				'timeout' => 0,
+				'headers' => [
+					'User-Agent' => 'Nextcloud Text',
+				],
+			];
+
+			if (count($params) > 0) {
+				if ($method === 'GET') {
+					$paramsContent = http_build_query($params);
+					$url .= '?' . $paramsContent;
+				} else {
+					$options['body'] = json_encode($params);
+				}
+			}
+
+			if ($method === 'GET') {
+				$response = $client->get($url, $options);
+			} elseif ($method === 'POST') {
+				$response = $client->post($url, $options);
+			} elseif ($method === 'PUT') {
+				$response = $client->put($url, $options);
+			} elseif ($method === 'DELETE') {
+				$response = $client->delete($url, $options);
+			} else {
+				return ['error' => 'Bad HTTP method'];
+			}
```

Comment:
There doesn't seem to be much use of other methods than GET here, right?

Reply:
Yes and we most likely won't use other methods later.